source/docs/xds.md

```diff
@@ -0,0 +1,44 @@
+# Envoy xDS subsystem
+
+## Overview
+
+Envoy implements the [xDS
+protocol](https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol). This document tracks
+some implementation details.
+
+An overview of the architecture is provided in the diagram below:
+
+![xDS implementation architecture overview](xDS_code_diagram.png)
+
+Note that this is currently somewhat idealized, with Envoy having two gRPC mux implementations,
+`GrpcMuxImpl` and `NewGrpxMuxImpl`, while we complete migration to a single muxer, see
+https://github.com/envoyproxy/envoy/issues/11477.
+
+The key components are:
+* `GrpcMuxImpl` which manages the xDS transport and multiplexing one or more subscriptions on a
+  single gRPC stream. The muxer is responsible for managing the wire level concerns via
+  `GrpcStream`, while also tracking subscription state and watchers for updates. Multiple resource
+  type URLs may be managed by a single muxer.
+* `SubscriptionState` tracks subscription state for a single resource type URL.
+* `WatchMap` tracks all the subscribers for a given resource type.
+* Each Envoy subsystem, e.g. LDS, CDS, etc. has a `GrpcSubscription` that points at a `GrpcMuxImpl`.
+
+## xdstp:// naming
```

Comment:
High level question: this will only be added to deltaXds, is that right? Do you want to make it clear here?

Reply:
Right, docs comment added.